### PR TITLE
use react-query and axios to call api

### PR DESCRIPTION
**Pull request by Tin-Cheng**
Install react-query and axios.
Call php api with react-query and axios instead of ajax in js.

### CODE REVIEW

File: client/src/components/App.js [hunk 1]
```diff
@@ -14,10 +14,55 @@ import oldTimetableScript from '../timetable';
...
         oldTimetableScript();
-    }, []);
+    }, [data]);
+    //TODO error handling
+    //if (isLoading) return "Loading...";
+    //if (error) return "An error has occurred: " + error.message;
+    
+    //TODO mirgate from timetable.js
+    //getURL();
+    //$("#add").autocomplete();
```
Comment (antonysiuming) [inline client/src/components/App.js:65]: Thanks for the contribution.

I think it is better to remove the unused code despite it is next to a TODO.

File: client/src/index.js [hunk 2]
```diff
@@ -4,15 +4,19 @@ import { Provider } from 'react-redux';
 import './index.css';
 import App from './components/App';
 import * as serviceWorker from './serviceWorker';
+import { QueryClient, QueryClientProvider } from "react-query";
```
Comment (antonysiuming) [inline client/src/index.js:7]: Please help to put the NPM package imports in the top.

**Review by antonysiuming — CHANGES_REQUESTED** [review]

File: client/src/components/App.js [hunk 3]
```diff
@@ -14,10 +14,55 @@ import oldTimetableScript from '../timetable';
...
+    data: "", // data get from courseInfo.json (via data.php and .json updated by mkdata.php)
+    loaded: false, // check if data loaded when adding course
+    searchHints: [],
+    semester: null, // store term in use
+    timetable: {}, // store the timetable
+  
+    API_PATH: 'https://coust.442.hk/',
+    CLIENT_PATH: 'https://coust.github.io/',
+    COOKIE_EXPIRE_DAYS: 50,
+  };
```
Comment (antonysiuming) [inline client/src/components/App.js:34]: Remove this first as this is WIP

**Review by antonysiuming — CHANGES_REQUESTED** [review]

File: client/src/components/App.js (hunk 1 above)
Comment (Tin-Cheng) [reply client/src/components/App.js:65]: Ok I will fix it.

**Review by antonysiuming — APPROVED** [review]
Thanks 💯

**Pull request merged**